ckanext/data_qld/de_identified_data/helpers.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from ckan import authz
+
+log = logging.getLogger(__name__)
+
+
+def process_de_identified_data_dict(data_dict, user_obj):
+    """
+    Remove de_identified_data value from dict
+    if current user doesn't have access to it.
+    """
+    # Add default value to NO.
+    if data_dict.get('de_identified_data', None) is None:
+        data_dict['de_identified_data'] = 'NO'
+
+    if not show_de_identified_data(data_dict, user_obj):
+        del data_dict['de_identified_data']
+
+
+def user_has_adm_editor_org_access(group_id, user_id):
+    """
+    Return False if the user does not have
+    admin, editor, or sysadmin access to the datasets organisation.
+    """
+    role = authz.users_role_for_group_or_org(group_id, user_id)
```

Comment:
What happens if the user has a member role for this organisation and a editor/admin role but the member role is returned from this method?
I think we should use this method 'authz.has_user_permission_for_group_or_org(org_id, user_name, 'create_dataset')' and use the 'create_dataset' permission.

Reply:
refactored to use `has_user_permission_for_group_or_org` instead,

it should return 'member' then, and the method will return false.